packages/rest/src/request-context.ts

```diff
@@ -95,7 +95,9 @@ export class RequestContext extends Context implements HandlerContext {
     super(parent, name);
     this._setupBindings(request, response);
     onFinished(this.response, () => {
-      this.close();
+      // See https://github.com/strongloop/loopback-next/issues/2541
+      // We call `clearSubscriptions` instead of `close`
+      this.clearSubscriptions();
```

Comment:
Please add an acceptance-level test demonstrating a somewhat realistic scenario, the test should be failing against the current master and pass with your changes in place.